npm_and_yarn/lib/dependabot/npm_and_yarn/package_manager.rb

```diff
@@ -53,7 +92,7 @@ def install(name, version)
       def requested_version(name)
         return unless @package_manager
 
-        match = @package_manager.match(/#{name}@(?<version>\d+.\d+.\d+)/)
+        match = @package_manager.match(/^#{name}@(?<version>\d+.\d+.\d+)/)
```

Comment:
Any reason for this regex change?

Reply:
@abdulapopoola , there was a match issue between npm and pnpm. "p**npm**" has substring "npm" . So made the change to match exact engine names.